### ISSUE TITLE
PropTypes errors when upgrading from 6.0 to 6.1

### ISSUE THREAD

**Issue by KevinGhadyani-minted** [storybookjs/storybook#13252]
**Describe the bug**
Seeing a prop types error when upgrading from 6.0 to 6.1.

I tested `6.0.22` and `6.0.28` with no problems. It was only when I went to `6.1.x` that I started seeing issues.

**To Reproduce**

1. Upgrade to Storybook `6.1.x` (tested: `6.1.0`, `6.1.1`, `6.1.2`, `6.1.3`, `6.1.5`, `6.1.9`, and `6.1.11`).
1. Open Storybook with this particular component.
1. See error.

**Expected behavior**
I expect to see no error.

**Screenshots**
![image](https://user-images.githubusercontent.com/45860344/100136980-5ea48100-2e51-11eb-8da7-d5ed2bc9c090.png)

**Code snippets**
`Occasions.js`
```js
export const minimumHeightDisplayModes = {
  CONTENT_HEIGHT: 'CONTENT_HEIGHT',
  MINIMUM_HEIGHT: 'MINIMUM_HEIGHT',
  NO_MINIMUM: 'NO_MINIMUM',
};

const Occasions = ({
  // ...
  minimumHeightDisplayMode = (
    minimumHeightDisplayModes
    .CONTENT_HEIGHT
  ),
  // ...
}) => {
  // ...
}
```

`Occasions.stories.js`
```js
export default {
  argTypes: {
    // ...
    minimumHeightDisplayMode: {
      // ...
      control: {
        options: (
          Occasions
          .minimumHeightDisplayModes
        ),
        type: 'select',
      },
      table: {
        defaultValue: {
          summary: (
            Occasions
            .minimumHeightDisplayModes
            .CONTENT_HEIGHT
          ),
        },
        type: {
          summary: 'string',
        },
      },
    },
  }
}
```

**System**
Docker container in the working state:
```
  System:
    OS: Linux 5.4 Alpine Linux
    CPU: (4) x64 AMD Ryzen 7 3800X 8-Core Processor
  Binaries:
    Node: 12.19.1 - /usr/local/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  npmPackages:
    @storybook/addon-actions: 6.0.28 => 6.0.28
    @storybook/addon-essentials: 6.0.28 => 6.0.28
    @storybook/addon-links: 6.0.28 => 6.0.28
    @storybook/react: 6.0.28 => 6.0.28
```

**Additional context**
N/A

**Comment by KevinGhadyani-minted** [comment]
I am still seeing these errors in `6.1.9`.

**Comment by KevinGhadyani-minted** [comment]
Figured out how to reproduce this issue. Looks like it's related to creating an enum-style object like so:
```js
export const loadingStates = {
  LOADED: 'LOADED',
  LOADING: 'LOADING',
};
```

Then assign that as a default value to a component prop:
```js
const MyComponent = ({
  loadingState: loadingStates.LOADING,
}) => {
  // ...
}
```

This also happens when passed in as a Storybook default value:
```js
export default {
  argTypes: {
    loadingState: {
      table: {
        defaultValue: {
          summary: (
            loadingStates.LOADING
          ),
        },
        type: {
          summary: 'string',
        },
      },
    },
  }
}
```

I think Storybook is string-parsing the value rather than using a string representation of it.

It's assuming that if you say "this is a string value", that you passed in a string. Instead, I passed in a computed string off an object prop.

**Comment by KevinGhadyani-minted** [comment]
Just tried it again with the latest `6.1.11`, same issue:

![image](https://user-images.githubusercontent.com/45860344/103688952-3a5fb800-4f58-11eb-9ba6-406e6f9e70d0.png)

What changed in 6.1? Is Storybook now string-parsing or AST-parsing my prop types? Why does it work at runtime and in 6.0, but suddenly 6.1 breaks because I'm defaulting props from an object?

**Comment by KevinGhadyani-minted** [comment]
This issue is coming from the `Occasions` component where it's using the default value in the component _rather_ than the one passed into Storybook. That means Storybook is passing `undefined` for the `minimumHeightDisplayMode` prop.

Instead of Storybook's default used, this is the default that's being used:
```js
const Occasions = ({
  // ...
  minimumHeightDisplayMode = (
    minimumHeightDisplayModes
    .CONTENT_HEIGHT
  ),
  // ...
}) => {
  // ...
}
```

This means Storybook must be string-parsing this value and passing it in. Has to be something weird like that because otherwise, I wouldn't see this error. Clearly, this is not a string, it should be the value of an object property.

**Comment by KevinGhadyani-minted** [comment]
After some more research, it appears I can get this first error to appear by exporting the object used (`minimumHeightDisplayMode`) from another file as you can see here:

![image](https://user-images.githubusercontent.com/45860344/103692038-cffd4680-4f5c-11eb-95bb-0f330062f2c5.png)

**Comment by KevinGhadyani-minted** [comment]
The issue is the import. Once I remove the import, it's fixed. So if I define this object of possible props values in another place, that causes the issue. _**Why?**_ This is super strange.

**Comment by shilman** [comment]
The reason is because `react-docgen` is a static analysis that runs file by file, so it has problems with external references.

**Comment by KevinGhadyani-minted** [comment]
That makes a lot of sense. I dunno why I didn't think about static analysis. Is there another way to fix this? I'm assuming I'm not the only one wanting to do stuff like that where values come from another location.

Stringifying it for now is fine as a hack, but it's not representative of how someone else would use that component.

**Comment by shilman** [comment]
This underlying problem has shown up in a bunch of issues so I've documented the potential fix in #14092 and am closing this as a duplicate
